pkg/common/oras/authprovider/authprovider_test.go

```diff
@@ -60,6 +60,38 @@ func (ap *TestAuthProvider) Provide(_ context.Context, _ string) (AuthConfig, er
 	}, nil
 }
 
+func TestProvide_CreationOfAuthProvider_ExpectedResults(t *testing.T) {
+	var testProviderFactory defaultProviderFactory
+	// unsupported type for unmarshal
+	invalidConfig1 := map[string]interface{}{
```

Comment:
maybe you could try table driven test like this: https://github.com/deislabs/ratify/blob/main/pkg/verifier/notation/notation_test.go#L141